### DISCUSSION TITLE
Set default workflow permissions for dependabot through settings?

### DISCUSSION THREAD

**Discussion by nadav-lord** [community/community#121941] [Actions]
### Select Topic Area

Product Feedback

### Body

Hey all. In our organization we make a fairly extensive use of Github Actions for our CI/CD flows, and we set our workflow permissions through the org / repo settings. However, we also use dependabot which automatically opens PRs which go through our CI. These CI workflows may fail because by default the dependabot permissions are read-only and some actions may need write permissions. There are at least a couple of ways to circumvent this, such as adding permissions explicitly to the workflow or setting the token of the action explicitly, but we were wondering if it's not worth it to be able to set dependabot's default workflow permissions through the org / repo settings just the way you set them in the regular default workflow permissions section under Actions settings.

**Comment by Shafin5556** [comment]
Thanks for bringing up this issue. It's definitely a pain point we've experienced as well. Your suggestion to include dependabot's default workflow permissions in the org/repo settings makes a lot of sense. It would streamline our setup and prevent the headache of manually adjusting permissions for dependabot-triggered PRs. I'll make sure to bring this up with our team and see if we can advocate for this enhancement with Github. Appreciate your insights
